packages/client/lib/sync/skeleton.ts

```diff
@@ -252,17 +249,85 @@ export class Skeleton extends MetaDBManager {
       }
       return true
     }
+    if (force) {
+      lastchain.head = number
+      this.config.logger.debug(
+        `Beacon chain extended new head=${lastchain.head} tail=${lastchain.tail} next=${short(
+          lastchain.next
+        )}`
+      )
+    }
+    return false
+  }
 
-    // Update the database with the new sync stats and insert the new
-    // head header. We won't delete any trimmed skeleton headers since
-    // those will be outside the index space of the many subchains and
-    // the database space will be reclaimed eventually when processing
-    // blocks above the current head.
-    await this.putBlock(head)
-    lastchain.head = number
-    await this.writeSyncStatus()
+  /**
+   * Announce and integrate a new head.
+   * @returns if the new head causes a reorg.
```

Comment:
Should this return `true` on a reorg?  I'm assuming we can't return undefined here since the type is a `Promise<boolean>`

Can you also define what each of the function parameters do?  It's not obvious to me what `force`/`init` do.

Reply:
added